test/cli.js

```diff
@@ -301,6 +303,88 @@ describe('cli', function() {
     });
   });
 
+  describe('node-sass sass/ --output css/', function() {
+    it('should create the output directory', function(done) {
+      var src = fixture('input-directory/sass');
+      var dest = fixture('input-directory/css');
+      var bin = spawn(cli, [src, '--output', dest]);
+
+      bin.once('close', function() {
+        assert(fs.existsSync(dest));
+        rimraf.sync(dest);
+        done();
+      });
+    });
+
+    it('it should compile all files in the folder', function(done) {
+      var src = fixture('input-directory/sass');
+      var dest = fixture('input-directory/css');
+      var bin = spawn(cli, [src, '--output', dest]);
+
+      bin.once('close', function() {
+        var files = fs.readdirSync(dest).sort();
+        assert.deepEqual(files, ['one.css', 'two.css', 'nested'].sort());
+        var nestedFiles = fs.readdirSync(path.join(dest, 'nested'));
+        assert.deepEqual(nestedFiles, ['three.css']);
+        rimraf.sync(dest);
+        done();
+      });
+    });
+
+    it('should skip files with an underscore', function(done) {
+      var src = fixture('input-directory/sass');
+      var dest = fixture('input-directory/css');
+      var bin = spawn(cli, [src, '--output', dest]);
+
+      bin.once('close', function() {
+        var files = fs.readdirSync(dest);
+        assert.equal(files.indexOf('_skipped.css'), -1);
+        rimraf.sync(dest);
+        done();
+      });
+    });
+
+    it('should ignore nested files if --recursive false', function(done) {
+      var src = fixture('input-directory/sass');
+      var dest = fixture('input-directory/css');
+      var bin = spawn(cli, [
+        src, '--output', dest,
+        '--recursive', false
+      ]);
+
+      bin.once('close', function() {
+        var files = fs.readdirSync(dest);
+        assert.deepEqual(files, ['one.css', 'two.css']);
+        rimraf.sync(dest);
+        done();
+      });
+    });
+
+    it('should error if no output directory is provided', function(done) {
+      var src = fixture('input-directory/sass');
+      var bin = spawn(cli, [src]);
+      var msg = 'Specify an output directory when using multi-file compilation';
+
+      bin.stderr.once('data', function(data) {
+        assert(data.toString('utf8').indexOf(msg) > -1);
+        done();
+      });
+    });
+
+    it('should error if output directory is not a directory', function(done) {
+      var src = fixture('input-directory/sass');
+      var dest = fixture('input-directory/sass/one.scss');
+      var bin = spawn(cli, [src, '--output', dest]);
+      var msg = util.format('Multi-file compilation: requires destination %s to be a directory.', dest);
+
+      bin.stderr.once('data', function(data) {
+        assert(data.toString('utf8').indexOf(msg) > -1);
```

Comment:
I tend to agree with @wesleytodd . I will rewrite these to check that the compiled target doesn't exist after running and probably that it threw an error. Because if it silently just failed that would also be bad.